test/evp_extra_test.c

```diff
@@ -2671,19 +3620,22 @@ static int test_set_get_raw_keys_int(int tst, int pub, int uselibctx)
          * Test that supplying a buffer that is too small fails. Doesn't apply
          * to HMAC with a zero length key
          */
-        if ((!pub && !TEST_false(EVP_PKEY_get_raw_private_key(pkey, buf,
+        if ((!pub && !TEST_false(EVP_PKEY_get_raw_private_key(pkey, shortbuf,
                                                                  &shortlen)))
-                || (pub && !TEST_false(EVP_PKEY_get_raw_public_key(pkey, buf,
+                || (pub && !TEST_false(EVP_PKEY_get_raw_public_key(pkey, shortbuf,
                                                                    &shortlen))))
             goto done;
     }
+    if (!TEST_ptr(buf = OPENSSL_zalloc(len <= 80 ? 80 : len)))
```

Comment:
Not sure why we need this condition on `len <= 80`

Reply:
The original code had a static 80 byte buffer, and was using it even in the `len == 0` case.
Switching to dynamic buffers to accomodate the larger ML-KEM keys I need to still use a non-NULL buffer even when `len` is zero. So I allocate at least 80 bytes.